### PR TITLE
refactor: refactor checker inbox & task to jetpack compose with multi module

### LINKED ISSUE #2082: refactor: migrate Checker Inbox & Tasks screen to jetpack compose with multi module
**Summary:** 

Transitioning from XML to Jetpack Compose offers advantages such as adopting a declarative UI approach, ensuring type-safe UI development, and minimizing boilerplate code.

**Select Type:**

- [ ] UI Improvements
- [ ] UI Add On
- [ ] Improve Existing Code
- [x] New Feature
- [ ] Other

### PR DESCRIPTION
**Pull request by Aditya-gupta99**
Fixes #2082 

<center>
    <table>
        <tr>
            <td align="center" valign="center">
<br><center>XML Screen</center>
<video src="https://github.com/openMF/android-client/assets/94394661/7ab0ff3c-0046-4a9a-b868-18494144d84f"> 
            </td>
            <td align="center" valign="center">
<br><center>Jetpack Compose</center>
<video src="https://github.com/openMF/android-client/assets/94394661/5a57df18-fae0-4866-b6f2-510b923c91fe">                 
                
            </td>
        </tr>
    </table>
</center>

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: feature/checker-inbox-task/src/main/java/com/sparklead/feature/checker_inbox_task/checker_inbox_and_task/presentation/CheckerInboxTasksScreen.kt [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+import com.sparklead.feature.checker_inbox_task.R
+
+/**
+ * Created by Aditya Gupta on 21/03/24.
+ */
+
+@Composable
+fun CheckerInboxTasksScreen(onBackPressed: () -> Unit, checkerInbox: () -> Unit) {
+
+    val checkerInboxTasksViewModel: CheckerInboxTasksViewModel = hiltViewModel()
```
Comment (therajanmaurya) [inline feature/checker-inbox-task/src/main/java/com/sparklead/feature/checker_inbox_task/checker_inbox_and_task/presentation/CheckerInboxTasksScreen.kt:55]: @Aditya-gupta99 read this https://medium.com/androidbytesensei/write-testable-compose-function-a381330b7d15 and inject viewmodel as hilt in function directly

File: feature/checker-inbox-task/src/main/java/com/sparklead/feature/checker_inbox_task/checker_inbox_and_task/presentation/CheckerInboxTasksScreen.kt [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    LaunchedEffect(key1 = true) {
+        checkerInboxTasksViewModel.loadCheckerTasksBadges()
+    }
+
+
+    Scaffold(
+        modifier = Modifier.fillMaxSize(),
+        containerColor = White,
+        topBar = {
+            TopAppBar(
```
Comment (therajanmaurya) [inline feature/checker-inbox-task/src/main/java/com/sparklead/feature/checker_inbox_task/checker_inbox_and_task/presentation/CheckerInboxTasksScreen.kt:69]: Move this in designsystem module and if we don't have generic class in designsystem them create one

File: feature/checker-inbox-task/src/main/java/com/sparklead/feature/checker_inbox_task/checker_inbox_and_task/presentation/CheckerInboxTasksScreen.kt [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+                            badge = state.checkerInboxBadge
+                        ) {
+                            checkerInbox()
+                        }
+                        TaskOptions(
+                            leadingIcon = R.drawable.ic_supervisor_account_24dp,
+                            option = "Client Approval",
+                            badge = "0"
+                        ) {
+
```
Comment (therajanmaurya) [inline feature/checker-inbox-task/src/main/java/com/sparklead/feature/checker_inbox_task/checker_inbox_and_task/presentation/CheckerInboxTasksScreen.kt:127]: why onclick is missing here?

File: feature/checker-inbox-task/src/main/java/com/sparklead/feature/checker_inbox_task/checker_inbox_and_task/presentation/CheckerInboxTasksScreen.kt [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+/**
+ * Created by Aditya Gupta on 21/03/24.
+ */
+
+@Composable
+fun CheckerInboxTasksScreen(onBackPressed: () -> Unit, checkerInbox: () -> Unit) {
+
+    val checkerInboxTasksViewModel: CheckerInboxTasksViewModel = hiltViewModel()
+    val state = checkerInboxTasksViewModel.checkerInboxTasksUiState.collectAsState().value
```
Comment (therajanmaurya) [inline feature/checker-inbox-task/src/main/java/com/sparklead/feature/checker_inbox_task/checker_inbox_and_task/presentation/CheckerInboxTasksScreen.kt:56]: Don't use collectAsState().value instead use `collectAsStateWithLifecycle()` every where

File: feature/checker-inbox-task/src/main/java/com/sparklead/feature/checker_inbox_task/checker_inbox_and_task/presentation/CheckerInboxTasksScreen.kt [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+    val isRefreshing by checkerInboxTasksViewModel.isRefreshing.collectAsState()
```
Comment (therajanmaurya) [inline feature/checker-inbox-task/src/main/java/com/sparklead/feature/checker_inbox_task/checker_inbox_and_task/presentation/CheckerInboxTasksScreen.kt:57]: Don't use collectAsState() instead use collectAsStateWithLifecycle() every where

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/checkerinbox/CheckerInboxTasksFragment.kt [hunk 6]
```diff
@@ -4,81 +4,39 @@ import android.os.Bundle
...
-    }
-
-    override fun onActivityCreated(savedInstanceState: Bundle?) {
-        super.onActivityCreated(savedInstanceState)
-        showMifosProgressBar()
-        viewModel = ViewModelProviders.of(
-            this, factory
-        ).get(CheckerInboxTasksViewModel::class.java)
-
-        viewModel.getCheckerTasks().observe(viewLifecycleOwner,
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/checkerinbox/CheckerInboxTasksFragment.kt:56]: I don't see this initial load of getCheckerTasks() and getRescheduleLoanTasks() in composable screen

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

File: feature/checker-inbox-task/src/main/java/com/sparklead/feature/checker_inbox_task/checker_inbox_and_task/presentation/CheckerInboxTasksScreen.kt (hunk 3 above)
Comment (Aditya-gupta99) [reply feature/checker-inbox-task/src/main/java/com/sparklead/feature/checker_inbox_task/checker_inbox_and_task/presentation/CheckerInboxTasksScreen.kt:98]: Till now there is no next screen for that feature.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/checkerinbox/CheckerInboxTasksFragment.kt (hunk 6 above)
Comment (Aditya-gupta99) [reply mifosng-android/src/main/java/com/mifos/mifosxdroid/online/checkerinbox/CheckerInboxTasksFragment.kt:56]: I have created one GetCheckerInboxBadgesUseCase which consist both.

**Review by therajanmaurya — APPROVED** [review]

**Pull request merged**